webui/react/src/hooks/usePolling.ts

```diff
@@ -21,42 +32,44 @@ const usePolling = (pollingFn: PollingFn, { delay }: PollingOptions = {}): StopF
     }
   }, []);
 
-  const poll = useCallback(async (): Promise<void> => {
-    await savedPollingFn.current();
+  const poll = useCallback(async () => {
+    if (stopPreviousPoll) clearTimer();
+    if (runImmediately) await savedPollingFn.current();
 
-    if (active.current) {
-      timer.current = setTimeout(() => {
-        timer.current = undefined;
-        poll();
-      }, delay || DEFAULT_DELAY);
-    }
-  }, [ delay ]);
+    timer.current = setTimeout(async () => {
+      await savedPollingFn.current();
+      timer.current = undefined;
+      if (isPolling.current) poll();
+    }, interval);
+  }, [ clearTimer, interval, runImmediately, stopPreviousPoll ]);
 
   const startPolling = useCallback(() => {
-    clearTimer();
-    active.current = true;
+    isPolling.current = true;
     poll();
-  }, [ clearTimer, poll ]);
+  }, [ poll ]);
 
   const stopPolling = useCallback(() => {
-    active.current = false;
+    isPolling.current = false;
     clearTimer();
   }, [ clearTimer ]);
 
   // Update polling function if a new one is passed in.
   useEffect(() => {
     savedPollingFn.current = pollingFn;
-    if (!active.current) startPolling();
-  }, [ pollingFn, startPolling ]);
+  }, [ pollingFn ]);
 
   // Start polling when mounted and stop polling when umounted.
   useEffect(() => {
     startPolling();
     return () => stopPolling();
+    /*
```

Comment:
Maybe I'm wrong but that should still happen by adding:
- `stopPolling`: depending only on `clearTimer` which never changes
- `startPolling`: depending on the params passed to `usePolling` (never changing) and `clearTimer` which never changes?

Reply:
yep that's very true. I did make a minor tweak to make `startPolling` only reliant on references. The polling option is not stored in a `useRef` reference so it should not cause the `startPolling` useCallback to be redefined which in turn should not cause the `useEffect` to re-run also.